sw/device/lib/crypto/impl/rsa.c

```diff
@@ -122,6 +227,66 @@ otcrypto_status_t otcrypto_rsa_public_key_construct(
   return OTCRYPTO_OK;
 }
 
+otcrypto_status_t otcrypto_rsa_public_key_deconstruct(
+    otcrypto_unblinded_key_t *public_key, otcrypto_word32_buf_t modulus,
+    uint32_t *exponent) {
+  if (public_key == NULL || public_key->key == NULL || modulus.data == NULL ||
+      exponent == NULL) {
+    return OTCRYPTO_BAD_ARGS;
+  }
+  // Entropy complex must be initialized for `hardened_memcpy`.
+  HARDENED_TRY(entropy_complex_check());
```

Comment:
Since this function is for public keys, `hardened_memcpy` might be overkill; it's a lot slower than a normal copy and a lot of its defenses are to prevent side-channel leakage rather than only faults. I think it's fine to leave it in, but I'd maybe add a clarifying comment that it's for fault resistance just so future readers understand why it's there. I do think extra care with RSA public keys matters, because it's so easy to fault them into weak moduli.